File: ProjectEuler/1-10/project-euler-9.java
/* 
Description: Pythagorean triple is a set of
three natural numbers a < b < c for which
a^2 + b^2 = c^2. Given N check if there 
exists any pythagorean triplet for which a+b+c = N 
If no pythagorean triplet exists, return -1
Completed using Java 8
*/

import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static int pythagoreanTriple(int n)
    {
        int product = -1; 
        for(int a = 3; a < (int) (n/3 + 1); a++)
        {
            int b = (int) ((Math.pow(n,2) - (2*a*n) ) / (2*n - 2*a)); 
            int c = n - b -a; 
            if(Math.pow(a,2) + Math.pow(b,2) == Math.pow(c,2))
            {
                if(a*b*c > product) product = a*b*c;
            }
        }
        return product; 
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(int a0 = 0; a0 < t; a0++){
            int n = in.nextInt();
            System.out.println(pythagoreanTriple(n)); 
        }
    }
}

File: ProjectEuler/project-euler-9.py
""" 
Description: Pythagorean triple is a set of
three natural numbers a < b < c for which
a^2 + b^2 = c^2. Given N check if there 
exists any pythagorean triplet for which a+b+c = N 
If no pythagorean triplet exists, return -1
Completed using Python 3.6.5 
""" 


#!/bin/python3
import sys
def pythagoreanTriple(n):
    product = -1 
    for a in range(3, (n//3) + 1):
        b = (n**2 - 2*a*n) // (2*n - 2*a) 
        c = n - b - a
        if a*a + b*b == c*c:
            if a*b*c > product: 
                product = a*b*c 
    return product 

t = int(input().strip())
for a0 in range(t):
    n = int(input().strip())
    print(pythagoreanTriple(n))
